### PR TITLE
More tests

### PR DESCRIPTION
**Pull request by andorp**

### CODE REVIEW

File: tests/AulaTests.hs [hunk 1]
```diff
@@ -33,11 +33,21 @@ import Servant          as X
...
 import Persistent (mkMetaInfo)
 
 
+-- Be default, test cases are part of the smoke test suite.
+data TestSuite
+    = Large
+  deriving (Eq, Show)
+
+instance Tag TestSuite where
+    tagText = cs . show
```
Comment (fisx) [inline tests/AulaTests.hs]: default instance!  (-:

it requires default signature again though, so @andorp you are probably against it, and i don't insist.  just saying it'd be fun...

File: .travis/docker-build.sh [hunk 2]
```diff
@@ -1,17 +1,32 @@
...
 # Change to the source directory which is attacehed as docker volume
 cd /liqd/aula
 
-if [ -z "$1" ] ; then
-    NO_OF_CASES="100"
-else
-    NO_OF_CASES="$1"
-fi
+NO_OF_CASES="100"
+SIZE_OF_CASES="30"
```
Comment (fisx) [inline .travis/docker-build.sh]: Not rooted in this PR, but related: why don't we call these QC_MAX_SUCCESS, QC_MAX_SIZE, like on the command line?

File: .travis/docker-build.sh [hunk 3]
```diff
@@ -1,17 +1,32 @@
...
+	SIZE_OF_CASES="$2"
+	shift
+	;;
+	*)
+	;;
+esac
+shift
+done
+
+stack install --fast --test --test-arguments "--qc-max-success=${NO_OF_CASES} --qc-max-size=${SIZE_OF_CASES}" --coverage --allow-different-user --pedantic aula
```
Comment (fisx) [inline .travis/docker-build.sh]: I'm really not a fan of shell scripts, but I'm assuming this works...

File: Makefile [hunk 4]
```diff
@@ -32,7 +32,7 @@ stories: .phony aula.unregister
 
 # aware of aula and thentos sources
 sensei-full: .phony unregister-full
-	$(EXEC) sensei -j5 $(FULL_SOURCES) -optP-DDEVELOPMENT ./tests/Spec.hs $(SENSEI_DEFAULT_ARGS) $(SENSEI_ARGS)
+	$(EXEC) sensei -j5 $(FULL_SOURCES) -optP-DDEVELOPMENT ./tests/Spec.hs $(SENSEI_DEFAULT_ARGS) $(SENSEI_ARGS) --skip @Large
```
Comment (fisx) [inline Makefile]: can you also change the `sensei` rule?

File: docs/test-suites.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
+# Testing
+
+During the development if an issue occurs, we try to program the type system
+to reject codes which could lead to similar issues in the future. The second
```
Comment (fisx) [inline docs/test-suites.md]: "code" (singular)

File: docs/test-suites.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+defense line is to write quick check properties that checks against the
```
Comment (fisx) [inline docs/test-suites.md]: "check" (plural :-)

File: docs/test-suites.md [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+assumptions we have at the moment.
+
+As the test suite grows it is necessary to make a separation between the
+tests which we run during the development cycle and those tests that should
+be run on the continuous integration. We call the first one as **smoke
+suite**, the second one we can ommit the 'smoke' preffix, or it can be called the **large suite**.
```
Comment (fisx) [inline docs/test-suites.md]: "we call it smoke suite" (no "as"); "omit"; "prefix"; can you break the line one more time?

File: docs/test-suites.md [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+
+Test cases marked with `Large` tag in the test suite run only in the
+the large test suite, which should run on the CI. This test
+suite uses different parameters for the HSpec test runner, such as
+running more test cases, and run quickcheck test with larger size
+parameter. Also includes the test cases with the `Large` tag.
+
+## Future test suites
+
+In the future integration, stress, and UI based test could be written
```
Comment (fisx) [inline docs/test-suites.md]: "In the future, integration, stress, and UI based tests could be written ..."

File: tests/Test/Hspec/Missing.hs [hunk 9]
```diff
@@ -0,0 +1,16 @@
+{-# OPTIONS_GHC -Wall -Werror #-}
+
+{-# LANGUAGE OverloadedStrings #-}
+module Test.Hspec.Missing where
```
Comment (fisx) [inline tests/Test/Hspec/Missing.hs]: could you add a haddock comment for the module that links to `docs/test-suites.md`?

**Comment by fisx** [comment]
Some comments, but nothing deep, I am very happy with the fact that we are testing more forms now (and with the fact that that didn't reveal any bugs :-).

Could you give iti another round and merge if you agree with everything I wrote?

**Comment by andorp** [comment]
We just run the formAction and redirectOf in these tests. At least we can check if those point to valid paths.

**Comment by fisx** [comment]
Perfect!  Thanks!  (-:

**Pull request merged**